### PR TITLE
X3D Scripting to GVRMaterial, Lights, Transform

### PR DESCRIPTION
**Pull request by mwitchwilliams**
Squashed commit of the following:

commit c75346b35cc1e224edb9672c31622f5a1ec0bd3f
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Thu Nov 17 15:03:13 2016 -0800

    Regresion tests, removed commented-out code and log commands.

commit 5fb1104d1519bce81b0c4a68e774e0a89b801375
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Wed Nov 16 11:25:20 2016 -0800

    SFBool SFFloat added as variables initiated with new Scripts.
    Near ready release for internal testing.  Next push will
    have comments and log files removed.

commit 3d5fee7f04780f3593d610ced7a2a23db02541ec
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Tue Nov 15 16:24:53 2016 -0800

    Implemented initialize() method, a method that constructs
    X3D data types in SCRIPT and setting arguments to pass
    to Mozilla Rhino.

commit 0ca17b2e2d4915952b5e2a298c9dbe9340d94771
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Mon Nov 14 16:37:01 2016 -0800

    Fixed issue with rotations.  Validated demo's work with
    constructing X3D data type objects in GearVRinitJavaScript.

commit 1ad6a5f090e8c13c9d882f570a49f85966fc81cc
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Fri Nov 11 16:19:54 2016 -0800

    Moved ConvertVectorToQuaternion from X3Dobject to
    AnimationInteractivityMgr so it can be shared.
    Developing GearVRinitJavaScript function to be embedded
    into the Script method to construct objects needed to
    run Scripts.  Broke a few things.  Will fix Mon. Nov. 14

commit 88ef13f38b8730f3ef2aae0f3a5e36a7273335d3
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Wed Nov 9 16:08:00 2016 -0800

    Added code for SFRotation

commit e050325c66508c6f6d0e14117edab3b20f2af22b
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Mon Nov 7 14:03:04 2016 -0800

    New data structure for fields of Script Object.
    Added SFloat and SFTime X3D data structures

commit 139e547317db5992837509bfd65cc02b86c010b7
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Fri Nov 4 16:24:54 2016 -0700

    Script object fields now handle connections to
    defined objects.  Thus, only one ScriptObject per
    Script tags.

commit 2b936de7b3e617eba20f711237a12881e984f1c8
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Mon Oct 31 16:10:51 2016 -0700

    Added initialize() method to Scripting, requiring some code
    to be reorganized.  Added SFBool, modified other X3D data types.

commit 4f406d46f9b77729a6b49a1fb43c68a6d1c236e0
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Fri Oct 28 16:05:02 2016 -0700

    Added SFVec3f and SFRotation X3D data type classes.
    Supporting transformations through Script node.

commit 4907ac3c189908d3eb26818088ad3f6b03d616d3
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Thu Oct 27 16:38:09 2016 -0700

    Script code fixes to allow multiple scripts in .x3d file
    and Script to set multiple values.

commit 53f7e5ea9543213d0faad194659ad6d5a8699d9d
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Wed Oct 26 15:13:47 2016 -0700

    Script changes multiple objects from a single script.

commit d36975faaec282a7102042fe5a49a47a31b1b123
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Tue Oct 25 16:02:54 2016 -0700

    isOver and isActive working for an SFColor data type

commit 0f90d092b611c2679a2dfc00e7dc17a3d2b076dd
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Mon Oct 24 12:17:36 2016 -0700

    Added public getBindings() to GVRScriptFile to view values passed to
    and modified in the JavaScript file.

commit 98786012d657c0c528889bd8d3b7b795877baee2
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Fri Oct 21 16:30:29 2016 -0700

    Work-in-progress added scripting to the AnimationManager.
    Calling the script from AnimationManager upon TouchSensor.

commit d37e9d993460fac582b1836253e7266b49249c94
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Wed Oct 19 14:05:15 2016 -0700

    Work-in-progress for X3D script node and calling Mozilla Rhino

commit e0e2d373c10f710056dda4d9a885808759932105
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Tue Oct 11 15:33:23 2016 -0700

    Fix to enable ROUTE with TouchSensor to come before or after other ROUTEs
    in this animation and interactivity.

commit b21e0faf727247b6114750a47cbc48fb555f3989
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Fri Oct 7 16:06:12 2016 -0700

    Added the isOver functionality to BooleanToggle controlled by
    a Touch Sensor.

commit 0d18349db8b5fd79efd922a200c138f7d3654c97
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Fri Oct 7 12:58:35 2016 -0700

    Saving version of BooleanToggle that works for isOver events from
    Touch Sensor

commit 4e989877dee6462f8ae3cad9bf4da504bfe1e445
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Thu Oct 6 11:50:37 2016 -0700

    Turning on / off lights with TouchSensor

commit f3b0589aa996a2ac36dbc5cf15444e5bbc84c1f7
Author: mwilliamsSamsung <m1.williams@partner.samsung.com>
Date:   Wed Oct 5 14:39:38 2016 -0700

    Work-in-progress development of sensor turning on a light

### CODE REVIEW

**Comment by mwitchwilliams** [comment]
GearVRf-DCO-1.0-Signed-off-by: Mitch Williams
m1.williams@partner.samsung.com

**Comment by liaxim** [comment]
Says the branch has conflicts:
```
Conflicting files
GVRf/Framework/framework/src/main/java/org/gearvrf/x3d/AnimationInteractivityManager.java
GVRf/Framework/framework/src/main/java/org/gearvrf/x3d/InteractiveObject.java
GVRf/Framework/framework/src/main/java/org/gearvrf/x3d/X3Dobject.java
```

**Comment by liaxim** [comment]
It would be nice if you'd apply standard Android code formatting. To the new files for now.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/script/GVRScriptFile.java [hunk 1]
```diff
@@ -178,7 +181,7 @@ public boolean invokeFunction(String funcName, Object[] params) {
 
         String statement = getInvokeStatementCached(funcName, params);
 
-        Bindings localBindings = null;
+        //Bindings localBindings = null;
```
Comment (thomasflynn) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/script/GVRScriptFile.java]: remove this line

**Comment by thomasflynn** [comment]
ok, just did a skimming of the code so far.  had one comment.  remove commented out lines that are not needed.  Also, ditto to applying the code formatting:
https://resources.samsungdevelopers.com/Gear_VR/020_GearVR_Framework_Project/040_Get_Involved#Coding_Guidelines

**Comment by thomasflynn** [comment]
ran with the associated Javascript test scripts.  works as expected.

**Comment by thomasflynn** [comment]
ok, looks good to me now.  any other comments?

**Comment by liaxim** [comment]
Skimmed over - looks good to me too.

**Pull request merged**